libbeat/processors/add_kubernetes_metadata/indexers.go

```diff
@@ -248,3 +249,81 @@ func containerID(status PodContainerStatus) string {
 	}
 	return ""
 }
+
+// IPPortIndexer indexes pods based on all their host:port combinations
+type IPPortIndexer struct {
+	genMeta GenMeta
+}
+
+// NewIPPortIndexer creates and returns a new indexer for pod IP & ports
+func NewIPPortIndexer(_ common.Config, genMeta GenMeta) (Indexer, error) {
+	return &IPPortIndexer{genMeta: genMeta}, nil
+}
+
+// GetMetadata returns metadata for the given pod, if it matches the index
+func (h *IPPortIndexer) GetMetadata(pod *Pod) []MetadataIndex {
+	commonMeta := h.genMeta.GenerateMetaData(pod)
+	hostPorts := h.GetIndexes(pod)
+	var metadata []MetadataIndex
+
+	if pod.Status.PodIP == "" {
```

Comment:
If a pod does not have an IP address, it means no metadata is availabe? Should this return an error?

Reply:
We wouldn't be dealing with this error anyway, it could make sense to log this, but I think it's just defensive code